# Урок 5. Домашнее задание

## Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование

In [2]:
import numpy as np
import pandas as pd
from scipy import stats

### Задача 5.1

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

#### Решение:

Гипотеза $H_0$: $M(X) = 17$, 

Гипотеза $H_1$: $M(X) = 17.5$, 

Уровень значимости $\alpha = 0.025$ при правосторонней критической области $\Omega_\alpha = (t_{1 - \alpha}, \infty)$,

Дисперсия $D(X) = 4$, 

n = 100,

Из центральной предельной теоремы мы знаем как определяются $a$ и $\sigma$:

$a = M(X), \text{ } \sigma = \sqrt {\frac{D(X)}{n}}.$

Поскольку выборка из 100 шариков распределена нормально, можно воспользоваться правилом двух сигм: значения диаметра шариков с вероятностью 0.95 попадают в интервал

$\left( 17 - 2 \cdot \sqrt{\frac{D(X)}{n}}, 17 + 2 \cdot \sqrt{\frac{D(X)}{n}} \right).$

Для нашего выбора $\alpha$ интересует правосторонняя критическая область:

$\left( 17 + 2 \cdot \sqrt{\frac{D(X)}{n}}, \infty \right)$

Критическая область будет иметь вид:

$\left( 17.4, \infty \right)$

#### Ответ:

Так как выборочное среднее получилось 17.5 и оно попадает в критическую область, то утверждение производителя не подверждается.

### Задача 5.2

Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания __a__ с надежностью 0.95, если выборочное среднее равно 80, а объём равен 256.

#### Решение:

Средне квадратическое отклонение генеральной совокупности $\sigma = 16$, 

Объем выборки из генеральной совокупности $n = 256$,

Выборочное среднее $m = 80$,

Надежность = 0.95, т.е. можно использовать правило двух сигм,



Найдем квантиль __t-распределения__ ($t_{1 - \alpha/2, \: n - 1}$) с помощью методов модуля `scipy.stats`:

In [26]:
# n-1
t_obj = stats.t(256 - 1)

In [27]:
# 1 - alpha / 2
t = t_obj.ppf(1 - 0.05 / 2)
t

1.9693105698491928

Тогда по формуле:

$
P \left( 
\overline{X} - t_{1 - \alpha/2, \: n - 1} \cdot \dfrac{S}{\sqrt{n}} 
\leq M(X) \leq
\overline{X} + t_{1 - \alpha/2, \: n - 1} \cdot \dfrac{S}{\sqrt{n}} 
\right) = 1 - \alpha.
$,

доверительный интервал равен:

In [29]:
s = 16
n = 256
m = 80

(m - t * s / np.sqrt(n), m + t * s / np.sqrt(n))

(78.0306894301508, 81.9693105698492)

### Задача 5.3

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
```
202, 203, 199, 197, 195, 201, 200, 204, 194, 190
```
Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

#### Решение:

$
P \left( 
\overline{X} - t_{1 - \alpha/2, \: n - 1} \cdot \dfrac{S}{\sqrt{n}} 
\leq M(X) \leq
\overline{X} + t_{1 - \alpha/2, \: n - 1} \cdot \dfrac{S}{\sqrt{n}} 
\right) = 1 - \alpha.
$

In [43]:
n = 10
alpha = 0.01

samples = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

In [44]:
m = samples.mean()
m

198.5

In [45]:
s = samples.std(ddof=1)
s

4.453463071962462

In [46]:
# n-1
t_obj = stats.t(n - 1)

In [47]:
# 1 - alpha / 2
t = t_obj.ppf(1 - alpha / 2)
t

3.2498355440153697

In [48]:
(m - t * s / np.sqrt(n), m + t * s / np.sqrt(n))

(193.92322840036354, 203.07677159963646)

#### Ответ:

Утверждение продавца верно, т.к. оно попадает в доверительный интервал.

#### Комментарий:

Юрий, вы в телеграмме писали, что эта задача не на доверительные интервалы и приложили обновление. Что то я совсем не понял как решать по такой формуле:

$\Omega_\alpha = \left( -\infty, - t_{1 - \alpha/2, \: n - 1}, t_{1 - \alpha/2, \: n - 1}, \infty \right)$

По ней критическая область получается: 

$\Omega_\alpha = \left( -\infty, - 3.2498; 3.2498, \infty \right)$

И что в итоге это нам дает и как с этим работать - непонятно.